File: src/Scripts/OGCameraWindow.js
#pragma strict

@script AddComponentMenu ("OpenGUI/CameraWindow")

class OGCameraWindow extends OGWidget {
	var targetCamera : Camera;
	
	override function Draw ( x : float, y : float ) {
		if ( targetCamera ) {
			if (Event.current.type == EventType.Repaint) {
				var cWidth : float = transform.localScale.x / Screen.width;
				var cHeight : float = transform.localScale.y / Screen.height;
				var cPosX : float = x / Screen.width;
				var cPosY : float = 1 - ( y / Screen.height ) - cHeight;
				
				targetCamera.rect = new Rect ( cPosX, cPosY, cWidth, cHeight );
				
				targetCamera.Render();
			}
		}
	}
}

File: c#/Scripts/OGCameraWindow.cs
using UnityEngine;
using System.Collections;

[AddComponentMenu ("OpenGUI/CameraWindow")]
public class OGCameraWindow : OGWidget {

	public Camera targetCamera;
	
	public override void Draw (  float x ,   float y   ){
		if ( targetCamera ) {
			if (Event.current.type == EventType.Repaint) {
				float cWidth = transform.localScale.x / Screen.width;
				float cHeight = transform.localScale.y / Screen.height;
				float cPosX = transform.position.x / Screen.width;
				float cPosY = 1 - ( transform.position.y / Screen.height ) - cHeight;
				
				targetCamera.rect = new Rect ( cPosX, cPosY, cWidth, cHeight );
				
				targetCamera.Render();
			}
		}
	}
}
